### ISSUE TITLE
openswoole 22.1 pecl build fails in alpine based docker containers

### ISSUE THREAD

**Issue by dreadkopp** [openswoole/openswoole#67]
Currently trying to get images ready for php8.3

however i cannot get openswoole to build via pecl in alpine based image

debian.Dockerfile (works)

```
FROM php:8.3.0

RUN apt update  \
 && apt install -y libssl-dev libcurl4-openssl-dev \
 && pecl install -D 'enable-sockets="no" enable-openssl="yes" enable-http2="yes" enable-mysqlnd="no" enable-hook-curl="yes" enable-cares="yes" with-postgres="no"' openswoole-22.1.0 \
 && docker-php-ext-enable openswoole

ENTRYPOINT php -m | grep openswoole
```

alpine.Dockerfile (fails, see log)

```
FROM php:8.3.0-alpine3.18

RUN apk add --no-cache --virtual .build-deps $PHPIZE_DEPS \
 && apk add openssl-dev \
 && pecl install -D 'enable-sockets="no" enable-openssl="yes" enable-http2="yes" enable-mysqlnd="no" enable-hook-curl="yes" enable-cares="yes" with-postgres="no"' openswoole-22.1.0 \
 && apk del .build-deps

RUN php -m
```


Build Log:

```
[+] Building 12.9s (5/6)                                                                                                                                                                                                                                                                                                                             docker:default
 => [internal] load build definition from 8.3-alpine-swoole.Dockerfile                                                                                                                                                                                                                                                                                         0.0s
 => => transferring dockerfile: 388B                                                                                                                                                                                                                                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                              0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                                                                0.0s
 => [internal] load metadata for docker.io/library/php:8.3.0-alpine3.18                                                                                                                                                                                                                                                                                        0.0s
 => CACHED [1/3] FROM docker.io/library/php:8.3.0-alpine3.18                                                                                                                                                                                                                                                                                                   0.0s
 => ERROR [2/3] RUN apk add --no-cache --virtual .build-deps autoconf   dpkg-dev dpkg   file   g++   gcc   libc-dev   make   pkgconf   re2c  && apk add openssl-dev  && pecl install -D 'enable-sockets="no" enable-openssl="yes" enable-http2="yes" enable-mysqlnd="no" enable-hook-curl="yes" enable-cares="yes" with-postgres="no"' openswoole-22.1.0  &&  12.9s
------                                                                                                                                                                                                                                                                                                                                                              
 > [2/3] RUN apk add --no-cache --virtual .build-deps autoconf          dpkg-dev dpkg           file            g++             gcc             libc-dev                make            pkgconf                 re2c  && apk add openssl-dev  && pecl install -D 'enable-sockets="no" enable-openssl="yes" enable-http2="yes" enable-mysqlnd="no" enable-hook-curl="yes" enable-cares="yes" with-postgres="no"' openswoole-22.1.0  && apk del .build-deps:                                                                                                                                                                                                                                                                              
0.173 fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz                                                                                                                                                                                                                                                                                 
0.446 fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/x86_64/APKINDEX.tar.gz                                                                                                                                                                                                                                                                            
0.607 (1/27) Installing m4 (1.4.19-r3)                                                                                                                                                                                                                                                                                                                              
0.648 (2/27) Installing libbz2 (1.0.8-r5)
0.677 (3/27) Installing perl (5.36.2-r0)
0.879 (4/27) Installing autoconf (2.71-r2)
0.918 (5/27) Installing pkgconf (1.9.5-r0)
0.948 (6/27) Installing dpkg-dev (1.21.21-r1)
0.992 (7/27) Installing libmd (1.0.4-r2)
1.019 (8/27) Installing dpkg (1.21.21-r1)
1.057 (9/27) Installing libmagic (5.45-r0)
1.103 (10/27) Installing file (5.45-r0)
1.130 (11/27) Installing libstdc++ (12.2.1_git20220924-r10)
1.179 (12/27) Installing libstdc++-dev (12.2.1_git20220924-r10)
1.387 (13/27) Installing zstd-libs (1.5.5-r4)
1.423 (14/27) Installing binutils (2.40-r7)
1.482 (15/27) Installing libgomp (12.2.1_git20220924-r10)
1.516 (16/27) Installing libatomic (12.2.1_git20220924-r10)
1.548 (17/27) Installing gmp (6.2.1-r3)
1.579 (18/27) Installing isl26 (0.26-r1)
1.628 (19/27) Installing mpfr4 (4.2.0_p12-r0)
1.662 (20/27) Installing mpc1 (1.3.1-r1)
1.690 (21/27) Installing gcc (12.2.1_git20220924-r10)
2.184 (22/27) Installing musl-dev (1.2.4-r2)
2.248 (23/27) Installing libc-dev (0.7.2-r5)
2.274 (24/27) Installing g++ (12.2.1_git20220924-r10)
2.448 (25/27) Installing make (4.4.1-r1)
2.474 (26/27) Installing re2c (3.0-r1)
2.509 (27/27) Installing .build-deps (20231204.103422)
2.510 Executing busybox-1.36.1-r5.trigger
2.514 OK: 298 MiB in 64 packages
2.558 fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/x86_64/APKINDEX.tar.gz
2.823 fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/x86_64/APKINDEX.tar.gz
3.033 (1/1) Installing openssl-dev (3.1.4-r1)
3.076 OK: 301 MiB in 65 packages
5.585 downloading openswoole-22.1.0.tgz ...
5.585 Starting to download openswoole-22.1.0.tgz (1,252,350 bytes)
5.585 ........................................................................................................................................................................................................................................................done: 1,252,350 bytes
6.506 485 source files, building
6.507 running: phpize
6.510 Configuring for:
6.510 PHP Api Version:         20230831
6.510 Zend Module Api No:      20230831
6.510 Zend Extension Api No:   420230831
6.844 config.m4:327: warning: The macro `AC_PROG_CC_C99' is obsolete.
6.844 config.m4:327: You should run autoupdate.
6.844 ./lib/autoconf/c.m4:1659: AC_PROG_CC_C99 is expanded from...
6.844 config.m4:327: the top level
7.344 building in /tmp/pear/temp/pear-build-defaultuserBBAjlH/openswoole-22.1.0
7.344 running: /tmp/pear/temp/openswoole/configure --with-php-config=/usr/local/bin/php-config --enable-sockets=no --enable-openssl=yes --enable-http2=yes --enable-mysqlnd=no --enable-hook-curl=yes --with-postgres=no
7.396 checking for grep that handles long lines and -e... /bin/grep
7.396 checking for egrep... /bin/grep -E
7.397 checking for a sed that does not truncate output... /bin/sed
7.403 checking for pkg-config... /usr/bin/pkg-config
7.403 checking pkg-config is at least version 0.9.0... yes
7.403 checking for cc... cc
7.431 checking whether the C compiler works... yes
7.431 checking for C compiler default output file name... a.out
7.449 checking for suffix of executables...
7.469 checking whether we are cross compiling... no
7.482 checking for suffix of object files... o
7.493 checking whether the compiler supports GNU C... yes
7.506 checking whether cc accepts -g... yes
7.545 checking for cc option to enable C11 features... none needed
7.558 checking how to run the C preprocessor... cc -E
7.578 checking for icc... no
7.584 checking for suncc... no
7.584 checking for system library directory... lib
7.603 checking if compiler supports -Wl,-rpath,... yes
7.628 checking build system type... x86_64-pc-linux-musl
7.628 checking host system type... x86_64-pc-linux-musl
7.628 checking target system type... x86_64-pc-linux-musl
7.657 checking for PHP prefix... /usr/local
7.657 checking for PHP includes... -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib
7.657 checking for PHP extension directory... /usr/local/lib/php/extensions/no-debug-non-zts-20230831
7.658 checking for PHP installed headers prefix... /usr/local/include/php
7.666 checking if debug is enabled... no
7.674 checking if zts is enabled... no
7.674 checking for gawk... no
7.674 checking for nawk... no
7.674 checking for awk... awk
7.675 checking if awk is broken... no
7.675 checking enable debug log... no
7.675 checking enable trace log... no
7.675 checking enable sockets support... no
7.675 checking enable openssl support... yes
7.675 checking enable HTTP2 support... yes
7.675 checking openswoole support... yes, shared
7.675 checking enable mysqlnd support... no
7.675 checking enable PostgreSQL support... no
7.675 checking enable c-ares support... no
7.675 checking dir of openssl... no
7.675 checking dir of jemalloc... no
7.675 checking enable asan... no
7.675 checking whether to enable coverage testing... no
7.675 checking whether to enable developer build flags... no
7.675 checking whether to enable Coroutine Hooks for CURL build flags... yes
7.675 checking whether to enable thread context... no
7.683 checking if compiling with clang... no
7.707 checking for accept4 in -lc... yes
7.727 checking for signalfd in -lc... yes
7.747 checking for eventfd in -lc... yes
7.768 checking for epoll_create in -lc... yes
7.789 checking for poll in -lc... yes
7.808 checking for sendfile in -lc... yes
7.831 checking for kqueue in -lc... no
7.855 checking for backtrace in -lc... no
7.874 checking for daemon in -lc... yes
7.894 checking for mkostemp in -lc... yes
7.914 checking for inotify_init in -lc... yes
7.938 checking for malloc_trim in -lc... no
7.959 checking for inotify_init1 in -lc... yes
7.981 checking for ptrace in -lc... yes
8.000 checking for getrandom in -lc... yes
8.022 checking for arc4random in -lc... no
8.042 checking for pthread_rwlock_init in -lpthread... yes
8.063 checking for pthread_spin_lock in -lpthread... yes
8.083 checking for pthread_mutex_timedlock in -lpthread... yes
8.207 checking for pthread_barrier_init in -lpthread... yes
8.227 checking for pthread_mutexattr_setpshared in -lpthread... yes
8.249 checking for pthread_mutexattr_setrobust in -lpthread... yes
8.270 checking for pthread_mutex_consistent in -lpthread... yes
8.290 checking for pcre_compile in -lpcre... no
8.310 checking for ares_gethostbyname in -lcares... no
8.329 checking for gzgets in -lz... no
8.347 checking for BrotliEncoderCreateInstance in -lbrotlienc... no
8.355 checking for cpu affinity... no
8.371 checking for socket REUSEPORT... yes
8.379 checking for futex... no
8.397 checking for ucontext... yes
8.397 checking for g++... g++
8.418 checking whether the compiler supports GNU C++... yes
8.431 checking whether g++ accepts -g... yes
8.484 checking for g++ option to enable C++11 features... none needed
8.492 checking for valgrind... no
8.502 checking for stdio.h... yes
8.516 checking for stdlib.h... yes
8.530 checking for string.h... yes
8.545 checking for inttypes.h... yes
8.560 checking for stdint.h... yes
8.574 checking for strings.h... yes
8.589 checking for sys/stat.h... yes
8.606 checking for sys/types.h... yes
8.624 checking for unistd.h... yes
8.640 checking for struct cmsghdr... yes
8.663 checking for hstrerror... yes
8.686 checking for socketpair... yes
8.708 checking for if_nametoindex... yes
8.732 checking for if_indextoname... yes
8.752 checking for netdb.h... yes
8.771 checking for netinet/tcp.h... yes
8.788 checking for sys/un.h... yes
8.801 checking for sys/sockio.h... no
8.821 checking for field ss_family in struct sockaddr_storage... yes
8.836 checking if getaddrinfo supports AI_V4MAPPED... yes
8.852 checking if getaddrinfo supports AI_ALL... yes
8.864 checking if getaddrinfo supports AI_IDN... no
8.883 checking if gethostbyname2_r is supported... yes
8.891 checking for valgrind... no
8.912 checking for clock_gettime in -lrt... yes
8.946 checking openssl fallback dir... checking for SSL_connect in -lssl... yes
8.991 checking openswoole coverage... enabled
9.004 checking whether the compiler supports GNU C++... (cached) yes
9.004 checking whether g++ accepts -g... (cached) yes
9.004 checking for g++ option to enable C++11 features... (cached) none needed
9.040 checking how to run the C++ preprocessor... g++ -E
9.057 checking for a sed that does not truncate output... /bin/sed
9.061 checking for ld used by cc... /usr/x86_64-alpine-linux-musl/bin/ld
9.063 checking if the linker (/usr/x86_64-alpine-linux-musl/bin/ld) is GNU ld... yes
9.063 checking for /usr/x86_64-alpine-linux-musl/bin/ld option to reload object files... -r
9.064 checking for BSD-compatible nm... /usr/bin/nm -B
9.064 checking whether ln -s works... yes
9.064 checking how to recognize dependent libraries... pass_all
9.090 checking for dlfcn.h... yes
9.090 checking how to run the C++ preprocessor... g++ -E
9.105 checking the maximum length of command line arguments... 98304
9.142 checking command to parse /usr/bin/nm -B output from cc object... ok
9.143 checking for objdir... .libs
9.144 checking for ar... ar
9.144 checking for ranlib... ranlib
9.144 checking for strip... strip
9.186 checking if cc supports -fno-rtti -fno-exceptions... no
9.186 checking for cc option to produce PIC... -fPIC
9.196 checking if cc PIC flag -fPIC works... yes
9.215 checking if cc static flag -static works... yes
9.228 checking if cc supports -c -o file.o... yes
9.236 checking whether the cc linker (/usr/x86_64-alpine-linux-musl/bin/ld -m elf_x86_64) supports shared libraries... yes
9.250 checking whether -lc should be explicitly linked in... no
9.254 checking dynamic linker characteristics... GNU/Linux ld.so
9.254 checking how to hardcode library paths into programs... immediate
9.255 checking whether stripping libraries is possible... yes
9.255 checking if libtool supports shared libraries... yes
9.255 checking whether to build shared libraries... yes
9.256 checking whether to build static libraries... no
9.290 
9.290 creating libtool
9.316 appending configuration tag "CXX" to libtool
9.369 checking for ld used by g++... /usr/x86_64-alpine-linux-musl/bin/ld -m elf_x86_64
9.370 checking if the linker (/usr/x86_64-alpine-linux-musl/bin/ld -m elf_x86_64) is GNU ld... yes
9.374 checking whether the g++ linker (/usr/x86_64-alpine-linux-musl/bin/ld -m elf_x86_64) supports shared libraries... yes
9.394 checking for g++ option to produce PIC... -fPIC
9.404 checking if g++ PIC flag -fPIC works... yes
9.425 checking if g++ static flag -static works... yes
9.438 checking if g++ supports -c -o file.o... yes
9.438 checking whether the g++ linker (/usr/x86_64-alpine-linux-musl/bin/ld -m elf_x86_64) supports shared libraries... yes
9.438 checking dynamic linker characteristics... GNU/Linux ld.so
9.438 (cached) (cached) checking how to hardcode library paths into programs... immediate
9.531 configure: patching config.h.in
9.532 configure: creating ./config.status
9.551 config.status: creating config.h
9.574 running: make
9.605 /bin/sh /tmp/pear/temp/pear-build-defaultuserBBAjlH/openswoole-22.1.0/libtool --mode=compile g++ -I. -I/tmp/pear/temp/openswoole -I/tmp/pear/temp/pear-build-defaultuserBBAjlH/openswoole-22.1.0/include -I/tmp/pear/temp/pear-build-defaultuserBBAjlH/openswoole-22.1.0/main -I/tmp/pear/temp/openswoole -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -I/tmp/pear/temp/openswoole -I/tmp/pear/temp/openswoole/include -I/tmp/pear/temp/openswoole/ext-src -I/tmp/pear/temp/openswoole/thirdparty/hiredis  -DHAVE_CONFIG_H  -g -O2 -Wall -Wno-unused-function -Wno-deprecated -Wno-deprecated-declarations -std=c++11    -DENABLE_PHP_SWOOLE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/openswoole/ext-src/php_swoole.cc -o ext-src/php_swoole.lo  -MMD -MF ext-src/php_swoole.dep -MT ext-src/php_swoole.lo
9.650 mkdir ext-src/.libs
9.650  g++ -I. -I/tmp/pear/temp/openswoole -I/tmp/pear/temp/pear-build-defaultuserBBAjlH/openswoole-22.1.0/include -I/tmp/pear/temp/pear-build-defaultuserBBAjlH/openswoole-22.1.0/main -I/tmp/pear/temp/openswoole -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -I/tmp/pear/temp/openswoole -I/tmp/pear/temp/openswoole/include -I/tmp/pear/temp/openswoole/ext-src -I/tmp/pear/temp/openswoole/thirdparty/hiredis -DHAVE_CONFIG_H -g -O2 -Wall -Wno-unused-function -Wno-deprecated -Wno-deprecated-declarations -std=c++11 -DENABLE_PHP_SWOOLE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/openswoole/ext-src/php_swoole.cc -MMD -MF ext-src/php_swoole.dep -MT ext-src/php_swoole.lo  -fPIC -DPIC -o ext-src/.libs/php_swoole.o
10.09 In file included from /usr/include/c++/12.2.1/clocale:42,
10.09                  from /usr/include/c++/12.2.1/x86_64-alpine-linux-musl/bits/c++locale.h:41,
10.09                  from /usr/include/c++/12.2.1/bits/localefwd.h:40,
10.09                  from /usr/include/c++/12.2.1/string:43,
10.09                  from /tmp/pear/temp/openswoole/include/swoole_log.h:21,
10.09                  from /tmp/pear/temp/openswoole/include/swoole.h:129,
10.09                  from /tmp/pear/temp/openswoole/include/swoole_api.h:21,
10.09                  from /tmp/pear/temp/openswoole/ext-src/php_swoole_private.h:29,
10.09                  from /tmp/pear/temp/openswoole/ext-src/php_swoole_cxx.h:19,
10.09                  from /tmp/pear/temp/openswoole/ext-src/php_swoole.cc:16:
10.09 /tmp/pear/temp/openswoole/ext-src/php_swoole_private.h: In function 'char* php_swoole_http_build_query(zval*, size_t*, smart_str*)':
10.09 /tmp/pear/temp/openswoole/ext-src/php_swoole_private.h:1050:63: error: cannot convert 'std::nullptr_t' to 'size_t' {aka 'long unsigned int'}
10.09  1050 |         php_url_encode_hash_ex(HASH_OF(zdata), formstr, NULL, NULL, NULL, NULL, NULL, (int) PHP_QUERY_RFC1738);
10.09       |                                                               ^~~~
10.09       |                                                               |
10.09       |                                                               std::nullptr_t
10.09 In file included from /tmp/pear/temp/openswoole/ext-src/php_swoole_private.h:43:
10.09 /usr/local/include/php/ext/standard/php_http.h:25:64: note:   initializing argument 4 of 'void php_url_encode_hash_ex(HashTable*, smart_str*, const char*, size_t, const zend_string*, zval*, const zend_string*, int)'
10.09    25 |                                 const char *num_prefix, size_t num_prefix_len,
10.09       |                                                         ~~~~~~~^~~~~~~~~~~~~~
10.51 make: *** [Makefile:221: ext-src/php_swoole.lo] Error 1
10.53 ERROR: `make' failed
------
8.3-alpine-swoole.Dockerfile:3
--------------------
   2 |     
   3 | >>> RUN apk add --no-cache --virtual .build-deps $PHPIZE_DEPS \
   4 | >>>  && apk add openssl-dev \
   5 | >>>  && pecl install -D 'enable-sockets="no" enable-openssl="yes" enable-http2="yes" enable-mysqlnd="no" enable-hook-curl="yes" enable-cares="yes" with-postgres="no"' openswoole-22.1.0 \
   6 | >>>  && apk del .build-deps
   7 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c apk add --no-cache --virtual .build-deps $PHPIZE_DEPS  && apk add openssl-dev  && pecl install -D 'enable-sockets=\"no\" enable-openssl=\"yes\" enable-http2=\"yes\" enable-mysqlnd=\"no\" enable-hook-curl=\"yes\" enable-cares=\"yes\" with-postgres=\"no\"' openswoole-22.1.0  && apk del .build-deps" did not complete successfully: exit code: 1

```

**Comment by fpetrovic** [comment]
I have the same problem as well

**Comment by doubaokun** [comment]
Fixed in 22.1.1
